app/upgrades/v3rc5/testnet/upgrades.go

```diff
@@ -0,0 +1,42 @@
+package testnet
+
+import (
+	"context"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	"github.com/babylonlabs-io/babylon/v4/app/keepers"
+	"github.com/babylonlabs-io/babylon/v4/app/upgrades"
+	epochingkeeper "github.com/babylonlabs-io/babylon/v4/x/epoching/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+)
+
+func clearAllHistoricalEpochMsgs(ctx sdk.Context, k epochingkeeper.Keeper) error {
+	currentEpoch := k.GetEpoch(ctx).EpochNumber
+
+	// Clean all epoch messages before the current epoch
+	for epochNum := uint64(1); epochNum < currentEpoch; epochNum++ {
+		if k.GetQueueLength(ctx, epochNum) > 0 {
+			err := k.ClearEpochMsgs(ctx, epochNum)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func CreateUpgradeHandler() upgrades.UpgradeHandlerCreator {
```

Comment:
Is this PR also intended to be included in the v3 upgrade?
- If so, I think we need to add the cleaning logic to the v3 mainnet upgrade handler as well.
- If it’s meant for v4 instead, then we should add a v4 mainnet/testnet handler and include the logic there.